src/ui/common/src/components/pages/artifact/id/index.tsx

```diff
@@ -205,10 +205,69 @@ const ArtifactDetailsPage: React.FC<ArtifactDetailsPageProps> = ({
   const inputs = mapOperators([artifact.from]);
   const outputs = mapOperators(artifact.to ? artifact.to : []);
 
+  let upstream_pending = false;
+  inputs.some((operator) => {
```

Comment:
Just curious, why are you using some here instead of filter?

Reply:
I only want to check whether any of the upstream operators are pending. `some` terminates on first True whereas `filter` will create another array with all that matches the criteria. We might want to use `filter` in the future if we want to display something for the pending operators but for now, I am not doing that.